### PR TITLE
[INFINITY-2639] Add Zone/Region rules

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java [hunk 1]
```diff
@@ -209,28 +208,36 @@ private static PlacementRule parseRow(StringMatcher taskFilter, List<String> row
...
+                StringMatcher taskFilter,
+                String fieldName,
+                String operatorName,
+                Optional<String> ignoredParameter) {
+
+            switch (PlacementUtils.getField(fieldName)) {
+                case HOSTNAME:
+                    return new MaxPerHostnameRule(1, taskFilter);
+                case ATTRIBUTE:
+                default:
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java]: should the default case throw an `UnsupportedOperationException`?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java [hunk 2]
```diff
@@ -244,13 +251,21 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
...
-            if (isHostname(fieldName)) {
-                return HostnameRuleFactory.getInstance().require(ExactMatcher.create(parameter));
-            } else {
-                return AttributeRuleFactory.getInstance().require(ExactMatcher.createAttribute(fieldName, parameter));
+
+            switch (PlacementUtils.getField(fieldName)) {
+                case HOSTNAME:
+                    return HostnameRuleFactory.getInstance().require(ExactMatcher.create(parameter));
+                case ATTRIBUTE:
+                default:
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java]: should the default case throw an `UnsupportedOperationException`?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java [hunk 3]
```diff
@@ -285,10 +304,13 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
                         "Unable to parse max parameter as integer for '%s' operation: %s",
                         operatorName, parameter), e);
             }
-            if (isHostname(fieldName)) {
-                return new RoundRobinByHostnameRule(num, taskFilter);
-            } else {
-                return new RoundRobinByAttributeRule(fieldName, num, taskFilter);
+
+            switch (PlacementUtils.getField(fieldName)) {
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java]: should the default case throw an `UnsupportedOperationException`?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java [hunk 4]
```diff
@@ -301,13 +323,20 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
...
+                String fieldName,
+                String operatorName,
+                Optional<String> requiredParameter) throws IOException {
+
             String parameter = validateRequiredParameter(operatorName, requiredParameter);
-            if (isHostname(fieldName)) {
-                return HostnameRuleFactory.getInstance().require(RegexMatcher.create(parameter));
-            } else {
-                return AttributeRuleFactory.getInstance().require(RegexMatcher.createAttribute(fieldName, parameter));
+            switch (PlacementUtils.getField(fieldName)) {
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java]: should the default case throw an `UnsupportedOperationException`?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java [hunk 5]
```diff
@@ -319,13 +348,19 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
...
+                String fieldName,
+                String operatorName,
+                Optional<String> requiredParameter) throws IOException {
+
             String parameter = validateRequiredParameter(operatorName, requiredParameter);
-            if (isHostname(fieldName)) {
-                return HostnameRuleFactory.getInstance().avoid(RegexMatcher.create(parameter));
-            } else {
-                return AttributeRuleFactory.getInstance().avoid(RegexMatcher.createAttribute(fieldName, parameter));
+            switch (PlacementUtils.getField(fieldName)) {
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java]: should the default case throw an `UnsupportedOperationException`?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java [hunk 6]
```diff
@@ -348,24 +387,23 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
...
-            } else {
-                // Ensure that:
-                // - Task sticks to nodes with matching fieldName defined at all (AttributeRule)
-                // - Task doesn't exceed one instance on those nodes (MaxPerAttributeRule)
-                StringMatcher matcher = RegexMatcher.createAttribute(fieldName, ".*");
-                return new AndRule(
-                        AttributeRuleFactory.getInstance()
-                                .require(matcher), new MaxPerAttributeRule(max, matcher, taskFilter));
+
+            switch (PlacementUtils.getField(fieldName)) {
```
Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java]: should the default case throw an `UnsupportedOperationException`?

**Comment by gabrielhartmann** [comment]
@nickbp: Sorry for the confusion.  I added enum values prematurely.  The next PR will use them.

**Pull request merged**